docs/features/mcl.md

```diff
@@ -0,0 +1,85 @@
+# Setting up multi cluster support for KCC
+
+> NOTE: This feature is in alpha stage and is actively being worked on.
+
+Imagine we have two Kubernetes clusters, `mcl-cluster-1` and `mcl-cluster-2`. We want to run KCC in both for high availability, but only one instance should be the active leader at any time.
+
+### Setup
+
+#### 1. Cluster Lock: Powered GCS Bucket
+
```

Comment:
Please provide some context. You are creating the GCS Bucket to act as the backing store for the state of the leader election. This is a pluggable component ....